File: Frontend/src/types/template.ts
// Tipos para templates gerados pelo fluxo documental
// GeoLimites - Frontend

export interface Template {
  id: string;
  name: string;
  description?: string;
  fileUrl: string;
  filePath: string;
  memorialStandardId?: string;
  municipality?: string;
  abntNorm?: string;
  status: TemplateStatus;
  ownerId: string;
  ownerName?: string;
  createdAt: string;
  updatedAt: string;
}

export interface TemplateCreate {
  name: string;
  description?: string;
  fileUrl: string;
  filePath: string;
  memorialStandardId?: string;
  municipality?: string;
  abntNorm?: string;
  status?: TemplateStatus;
}

export interface TemplateUpdate {
  name?: string;
  description?: string;
  fileUrl?: string;
  filePath?: string;
  memorialStandardId?: string;
  municipality?: string;
  abntNorm?: string;
  status?: TemplateStatus;
}

export interface TemplateFormData {
  name: string;
  description: string;
  municipality: string;
  abntNorm: string;
  memorialStandardId: string;
  exampleFile?: File;
}

export type TemplateStatus = 'ACTIVE' | 'INACTIVE' | 'DRAFT';

export interface TemplateGenerationRequest {
  name: string;
  description?: string;
  municipality?: string;
  abntNorm?: string;
  memorialStandardId?: string;
  exampleFileId: string;
  targetFolderPath: string;
}

export interface TemplateGenerationResponse {
  id: string;
  name: string;
  fileUrl: string;
  filePath: string;
  message: string;
}

// Interface para seleção de templates (compatibilidade com código existente)
export interface TemplateOption {
  id: string;
  name: string;
  description?: string;
  file?: string;
}

File: Backend/src/main/java/com/momorialPro/CadMemorial/model/Template.java
package com.momorialPro.CadMemorial.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "tab_templates")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Template {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "file_url", nullable = false)
    private String fileUrl;

    @Column(name = "file_path", nullable = false)
    private String filePath;

    @Column(name = "memorial_standard_id")
    private UUID memorialStandardId;

    @Column
    private String municipality;

    @Column(name = "abnt_norm")
    private String abntNorm;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private TemplateStatus status = TemplateStatus.ACTIVE;

    @Column(name = "owner_id", nullable = false)
    private UUID ownerId;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    public enum TemplateStatus {
        ACTIVE, INACTIVE, DRAFT
    }
}